### PR TITLE
indexer: update elasticsearch to v7.5.0

### PR DESCRIPTION
**Pull request by pazembrz**
:snowflake: :santa: :christmas_tree: :snowman_with_snow: :gift: :mrs_claus: :snowflake:
* Updates ES in docker-compose
* Updates mappings
* Updated a lot of tests
* Updates kibana image in docker-compose

* Requires to reindex everything

* If you will have issues running new ES in docker add to sysctl.conf: vm.max_map_count = 262144 and run '# sysctl -p' to reload config

WARNING: ES now gives slightly changed 'results' object, now number of results are in 'results.total.value' but only when processing response directly from ES. In our API it's still 'results.total' as Invenio is serializing it and putting in an old format

* INSPIR-3040
* INSPIR-2467

### CODE REVIEW

File: backend/inspirehep/records/indexer/base.py [hunk 1]
```diff
@@ -22,14 +27,22 @@
...
     """Extend Invenio indexer to properly index Inspire records"""
 
-    def _prepare_record(self, record, index, doc_type):
+    @staticmethod
+    def _prepare_record(record, index, doc_type="_doc", arguments=None, **kwargs):
+        if doc_type and doc_type != "_doc":
+            warnings.warn(
+                "Argument doc_type is deprecated. Should be '_doc' or None.",
+                DeprecationWarning,
+            )
```
Comment (michamos) [inline backend/inspirehep/records/indexer/base.py]: do we really need deprecation warnings? this is not a library

File: backend/inspirehep/records/indexer/base.py [hunk 2]
```diff
@@ -90,6 +108,9 @@ def bulk_index(self, records_uuids, request_timeout=None):
             request_timeout=request_timeout,
             raise_on_error=False,
             raise_on_exception=False,
+            expand_action_callback=(
+                _es7_expand_action if ES_VERSION[0] >= 7 else default_expand_action
+            ),
```
Comment (michamos) [inline backend/inspirehep/records/indexer/base.py]: why? we are on ES 7.

File: backend/inspirehep/records/indexer/base.py [hunk 3]
```diff
@@ -57,6 +70,11 @@ def _process_bulk_record_for_index(
         index_from_record, doc_type_from_record = self.record_to_index(record)
         if not doc_type:
             doc_type = doc_type_from_record
+        elif doc_type and doc_type != "_doc":
+            warnings.warn(
+                "Argument doc_type is deprecated. Should be '_doc' or None.",
+                DeprecationWarning,
+            )
```
Comment (michamos) [inline backend/inspirehep/records/indexer/base.py]: again, why not simply remove all this and hardcode `_doc`?

File: backend/inspirehep/search/facets.py [hunk 4]
```diff
@@ -315,9 +316,10 @@ def citations_by_year():
...
-                    "reduce_script": "def results=[:]; for(result in params._aggs){for(key in result['_agg_yearly_cit'].keySet()){if(result['_agg_yearly_cit'][key] != null){results[key]=results.getOrDefault(key,0)+result['_agg_yearly_cit'][key]}}} return results",
+                    "init_script": "state._agg_yearly_cit=[:]",
+                    "map_script": "def years=params._source.citations_by_year != null ? params._source.citations_by_year : [] ; for(element in years){state._agg_yearly_cit[element.year.toString()]=state._agg_yearly_cit.getOrDefault(element.year.toString(), 0)+element.count}",
+                    "reduce_script": "def results=[:]; for(result in states){for(key in result['_agg_yearly_cit'].keySet()){if(result['_agg_yearly_cit'][key] != null){results[key]=results.getOrDefault(key,0)+result['_agg_yearly_cit'][key]}}} return results",
+                    "combine_script": "return state",
```
Comment (michamos) [inline backend/inspirehep/search/facets.py]: minor: swap these lines to make order more explicit

File: backend/inspirehep/search/mappings/v7/__init__.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+# flake8: noqa
+
+"""Mappings for Elasticsearch 5.x."""
```
Comment (michamos) [inline backend/inspirehep/search/mappings/v7/__init__.py]: wrong docstring

File: backend/inspirehep/search/mappings/v7/records/authors.json [hunk 6]
```diff
@@ -0,0 +1,399 @@
+{
```
Comment (michamos) [inline backend/inspirehep/search/mappings/v7/records/authors.json]: could you remove the v5 mappings? hopefully git will detect that the files have been moved and show a proper diff (unless there are large changes)

File: backend/tests/helpers/cleanups.py [hunk 7]
```diff
@@ -21,25 +21,31 @@ def es_cleanup(es):
...
+    from invenio_search import current_search
 
     es.indices.refresh()
-    existing_mappings = set(es.indices.stats()["indices"].keys())
+    existing_mappings = []
+    [
+        existing_mappings.extend(x["aliases"].keys())
+        for x in es.indices.get_alias("*").values()
+    ]
+    existing_mappings = set(existing_mappings)
```
Comment (michamos) [inline backend/tests/helpers/cleanups.py]: using comprehensions for side effects is not nice. I would use a for loop, or better, if you want to keep using a comprehension
```python
from itertools import chain
existing_mappings_nested = (x["aliases"].keys() for x in es.indises.get_alias("*").values())
existing_mappings = set(chain.from_iterable(existing_mappings_nested))
```

File: backend/tests/integration/records/serializers/json/test_authors.py [hunk 8]
```diff
@@ -22,7 +22,7 @@ def test_authors_detail(api_client, db, create_record_factory, datadir):
     record_control_number = record.json["control_number"]
 
     expected_status_code = 200
-    expected_id = record_control_number
+    expected_id = str(record_control_number)
```
Comment (michamos) [inline backend/tests/integration/records/serializers/json/test_authors.py]: I don't understand this. Let's discuss it IRL.

File: backend/tests/integration/records/serializers/json/test_literature.py [hunk 9]
```diff
@@ -624,7 +624,7 @@ def test_literature_references_with_size_bigger_than_maximum(
         )
     response_status_code = response.status_code
     response_data = json.loads(response.get_data())
-    expected_status_code = 400
+    expected_status_code = 200
```
Comment (michamos) [inline backend/tests/integration/records/serializers/json/test_literature.py]: why?

File: backend/tests/integration/records/views/test_views_jobs.py [hunk 10]
```diff
@@ -186,8 +186,6 @@ def test_jobs_record_search_results_returns_pending_job_to_superuser(
     expected_metadata.pop("_created")
     expected_results = 1
 
-    assert result.json["hits"]["total"] == expected_results
```
Comment (michamos) [inline backend/tests/integration/records/views/test_views_jobs.py]: why remove this?

File: backend/tests/integration/submissions/test_submissions_views.py [hunk 11]
```diff
@@ -164,7 +164,7 @@ def test_get_author_update_data(app, api_client, create_user, create_record_fact
     login_user_via_session(api_client, email=user.email)
 
     author_data = {
-        "control_number": 123,
+        "control_number": "123",
```
Comment (michamos) [inline backend/tests/integration/submissions/test_submissions_views.py]: schema says this is an integer

File: backend/tests/integration/records/views/test_views_jobs.py (hunk 10 above)
Comment (pazembrz) [reply backend/tests/integration/records/views/test_views_jobs.py]: the same check is on line 194 (192 after change) - 4 lines under this one

File: backend/inspirehep/search/mappings/v7/records/authors.json (hunk 6 above)
Comment (michamos) [reply backend/inspirehep/search/mappings/v7/records/authors.json]: To review the mappings, I checked out your commit, then did
```
git checkout master -- backend/inspirehep/search/mappings/v5
git diff -w --no-index inspirehep/search/mappings/v5/records/hep.json inspirehep/search/mappings/v7/records/hep.json
```
which gives a fairly useable diff.

File: backend/inspirehep/search/mappings/v7/records/hep.json [hunk 12]
```diff
@@ -0,0 +1,1279 @@
...
+                },
+                "type": "object"
+              },
+              "value": {
+                "fields": {
+                  "raw": {
+                    "type": "keyword"
+                  }
+                },
+                "normalizer": "lowercase_normalizer",
```
Comment (michamos) [inline backend/inspirehep/search/mappings/v7/records/hep.json]: copy to `_all` (you dropped it)

File: backend/tests/integration/search/test_factories_search.py [hunk 13]
```diff
@@ -115,7 +115,7 @@ def test_search_factory_with_aggs_with_query(base_app):
         "filters": {"type": mock_filter("type")},
         "post_filters": {"type": mock_post_filter("type")},
     }
-    config = {"RECORDS_REST_FACETS": {"_all": facets}}
+    config = {"RECORDS_REST_FACETS": {"*": facets}}
```
Comment (michamos) [inline backend/tests/integration/search/test_factories_search.py]: did you change this in the actual config?

File: backend/inspirehep/search/api.py [hunk 14]
```diff
@@ -43,9 +43,7 @@ def get_source(self, uuid, **kwargs):
         :type uuid: UUID
         :returns: dict
         """
-        return es.get_source(
-            index=self.Meta.index, doc_type=self.Meta.doc_types, id=uuid, **kwargs
```
[the remark this answers is not in the document]
Comment (michamos) [reply backend/inspirehep/search/api.py]: not possible, they're needed by invenio

File: backend/inspirehep/search/api.py [hunk 15]
```diff
@@ -208,38 +198,35 @@ def query_from_iq(self, query_string):
...
-            user_query = Q(IQ(query_string, self) & Q("term", status="open"))
+            if not query_string:
+                user_query = Q("term", status="open")
+            else:
+                user_query = Q(
+                    Q("query_string", query=query_string) & Q("term", status="open")
+                )
             return self.query(user_query)
-        return self.query(IQ(query_string, self))
+        return super().query_from_iq(query_string)
```
Comment (michamos) [inline backend/inspirehep/search/api.py]:
```suggestion
        return super().query_from_iq(user_query)
```
how are your new tests passing despite this?

File: backend/tests/integration/search/test_api.py [hunk 16]
```diff
@@ -30,3 +30,97 @@ def test_literature_get_records_by_pids_returns_correct_record(
...
     for rec in result:
         assert rec.to_dict()["control_number"] in expected_control_numbers
+
+
+def test_empty_literature_search(api_client, db, es_clear, create_record):
+    create_record("lit")
+    create_record("lit")
+    response = api_client.get("api/literature")
+
+    expectet_results_count = 2
```
Comment (michamos) [inline backend/tests/integration/search/test_api.py]:
```suggestion
    expected_results_count = 2
```
(this typo is present everywhere)

File: backend/inspirehep/search/api.py (hunk 15 above)
Comment (pazembrz) [reply backend/inspirehep/search/api.py]: No!

**Pull request merged**